src/components/KpiReact/index.js

```diff
@@ -0,0 +1,186 @@
+      let deceasedPercent = parseInt(
+        (totals.deceased / totals.confirmed) * 100
+      );
+      if (deceasedPercent < 1) {
+        deceasedPercent =
+          parseInt((totals.deceased / totals.confirmed) * 1000) / 10;
+      }
+      deaths.percentage = t("deceased-percentage", {
+        percent: formatNumber(deceasedPercent),
+      });
+    }
+    setDrawDeaths(deaths);
+  }, [data, language, t]);
+  useEffect(() => {
+    const { totals, totalsDiff } = data;
+    const formatNumber = maybeIntlNumberFormat(language);
+
+    let active = {
+      id: "active",
+      label: t("kpi-active"),
+      value: formatNumber(totals.active),
+      diff:
+        totalsDiff.active >= 0
+          ? `+${formatNumber(totalsDiff.active)}`
+          : formatNumber(totalsDiff.active),
+      chartName: "active_cumulative_avg.svg",
+      caption: t("active-chart-caption"),
+      isActive: true,
+    };
+    if (totals.active > 0) {
+      let criticalPercentage = parseInt(
+        (totals.critical / totals.active) * 100
+      );
+      if (criticalPercentage < 1) {
+        criticalPercentage =
+          parseInt((totals.critical / totals.active) * 1000) / 10;
+      }
+      active.percentage = t("active-critical-percentage", {
+        percent: formatNumber(criticalPercentage),
+      });
+    }
+
+    setDrawActive(active);
+  }, [data, language, t]);
+  useEffect(() => {
+    const { totals, totalsDiff } = data;
+    const formatNumber = maybeIntlNumberFormat(language);
+
+    let tested = {
+      id: "tested",
+      label: t("kpi-tested"),
+      value: formatNumber(totals.tested),
+      diff:
+        totalsDiff.tested >= 0
+          ? `+${formatNumber(totalsDiff.tested)}`
+          : formatNumber(totalsDiff.tested),
+      chartName: "tested_daily_avg.svg",
+      caption: t("tested-chart-caption"),
+    };
+
+    if (totals.tested > 0) {
+      let testedPercentage = parseInt((totals.confirmed / totals.tested) * 100);
+      if (testedPercentage < 1) {
+        testedPercentage =
+          parseInt((totals.confirmed / totals.tested) * 1000) / 10;
+      }
+      tested.percentage = t("tested-percentage", {
+        percent: formatNumber(testedPercentage),
+      });
+    }
+    setDrawTested(tested);
+  }, [data, language, t]);
+
+  const draw = (chart) => (
+    <Kpi
+      key={chart.id}
+      id={chart.id}
+      label={chart.label}
+      value={chart.value}
+      diff={chart.diff}
+      chartName={chart.chartName}
+      percent={chart.percentage}
+      caption={chart.caption}
+      isActive={chart.isActive}
+    />
+  );
+
+  switch (type) {
+    case "confirmed":
+      return drawConfirmed ? draw(drawConfirmed) : "";
+    case "recovered":
+      return drawRecovered ? draw(drawRecovered) : "";
+    case "deceased":
+      return drawDeaths ? draw(drawDeaths) : "";
+    case "active":
+      return drawActive ? draw(drawActive) : "";
+    case "tested":
+      return drawTested ? draw(drawTested) : "";
```

Comment:
If you move aggregating of chart data in a separate method you can get rid of this switch case and render Kpi component with correct data depends on a type